near-sdk-macros/src/core_impl/utils/mod.rs

```diff
@@ -39,3 +39,36 @@ pub(crate) fn extract_ok_type(ty: &Type) -> Option<&Type> {
         _ => None,
     }
 }
+
+/// Checks whether the given path is literally "Vec".
+/// Note that it won't match a fully qualified name `std::vec::Vec` or a type alias like
+/// `type MyVec = Vec<String>`.
+fn path_is_vec(path: &Path) -> bool {
+    path.leading_colon.is_none()
+        && path.segments.len() == 1
+        && path.segments.iter().next().unwrap().ident == "Vec"
```

Comment:
Why is this necessary? Anything else other than a `Vec` would fail to compile by the macros, so why not allow aliases? Very niche case so fine to ignore

Reply:
As I said in my other comment a verbose check just makes errors more transparent, but yeah this certainly comes with its own downsides. Happy to go either way.

Reply:
Yeah, we can leave for now. Easier to add support for aliases later than to take it away